src/Integration/RuleSetHelper.cs

```diff
@@ -62,6 +63,31 @@ public static IEnumerable<RuleSetInclude> FindAllIncludesUnderRoot(RuleSet ruleS
         }
 
         /// <summary>
+        /// Return a non-nested include from source to target
+        /// </summary>
+        /// <param name="source">Required</param>
+        /// <param name="target">Required</param>
+        /// <returns><see cref="RuleSetInclude"/> or null if not found</returns>
+        public static RuleSetInclude FindInclude(RuleSet source, RuleSet target)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (target == null)
+            {
+                throw new ArgumentNullException(nameof(target));
+            }
+
+            string relativeTargetFilePath = PathHelper.CalculateRelativePath(source.FilePath, target.FilePath);
+
+            return source.RuleSetIncludes.SingleOrDefault(i =>
+                StringComparer.OrdinalIgnoreCase.Equals(i.FilePath, relativeTargetFilePath)
```

Comment:
We could, the real question is whether we should? The code that generates the includes is out code, so it should be enough to assume and test that we do the right thing (we do and we do test).
I prefer to keep it simple in this case, rather than extensive on purpose, especially if the worst case will be a message prompting the user to update (which would only happen if have manually modified stuff).